pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -40,12 +42,18 @@ var _ model.Controller = &Controller{}
 type Controller struct {
 	registries []serviceregistry.Instance
 	storeLock  sync.RWMutex
+	meshHolder mesh.Holder
+}
+
+type Options struct {
+	MeshHolder mesh.Holder
 }
 
 // NewController creates a new Aggregate controller
-func NewController() *Controller {
+func NewController(opt *Options) *Controller {
```

Comment:
very nitpick: no need to make this a pointer

Reply:
Yeah and it makes it not clear why its a pointer: is nil valid? This is a super nitpick though